packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,156 @@
+  manyParticipantsChanged: string;
+  manyUnnamedParticipantsChanged: string;
+};
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const strings = useMemo(() => {
+    return {
+      participantJoinedNoticeString: locale.participantJoinedNoticeString,
+      twoParticipantJoinedNoticeString: locale.twoParticipantJoinedNoticeString,
+      threeParticipantJoinedNoticeString: locale.threeParticipantJoinedNoticeString,
+      participantLeftNoticeString: locale.participantLeftNoticeString,
+      twoParticipantLeftNoticeString: locale.twoParticipantLeftNoticeString,
+      threeParticipantLeftNoticeString: locale.threeParticipantLeftNoticeString,
+      unnamedParticipantChangedString: locale.unnamedParticipantChangedString,
+      manyParticipantsChanged: locale.manyParticipantsChanged,
+      manyUnnamedParticipantsChanged: locale.manyUnnamedParticipantsChanged
+    };
+  }, [locale]);
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipants = useSelector(getRemoteParticipantsConnectedSelector);
+  const [previousParticipants, setPreviousParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+
+    if (previousParticipants.length > remoteParticipants.length) {
+      const whoLeft = previousParticipants.filter((p) => !remoteParticipants.includes(p));
+      //someone left
+      setParticipantEventString(createAnnouncmentString('left', whoLeft, strings));
+      setPreviousParticipants(remoteParticipants);
+    } else if (remoteParticipants.length > previousParticipants.length) {
+      const whoJoined = remoteParticipants.filter((p) => !previousParticipants.includes(p));
+      // someone joined
+      setParticipantEventString(createAnnouncmentString('joined', whoJoined, strings));
+      setPreviousParticipants(remoteParticipants);
+    }
+  }, [remoteParticipants, previousParticipants, strings]);
+
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+export const createAnnouncmentString = (
+  direction: 'joined' | 'left',
+  participants: RemoteParticipantState[],
+  strings: ParticipantChangedAnnouncmentStrings
+): string => {
+  /**
+   * Check that we have more than 1 participant, if they all have no displayName return unnamed participants
+   * overflow string
+   */
+  if (participants.filter((p) => p.displayName).length === 0 && participants.length > 0) {
+    const participantsString = _formatString(strings.manyUnnamedParticipantsChanged, {
```

Comment:
I'd still push back on putting together strings from the locale like we do here.

It's clearer to add multiple strings here as well `manyUnnamedParticipantsJoined` `mainUnnamedParticipantsLeft`